### PR TITLE
feat(technicalkey): unexpected response for post policy-content - Purpose

### PR DESCRIPTION
**Pull request by AnuragNagpure**
## Description

Error message with http status code 400 that value is not allowed. Only backend stored purposes are allowed; any other should run on an error

## Why

The implementation should include the following requirements:

When the tester provides an input value that is not supported for the specified purposes, the endpoint should generate an error.
The error message displayed to the user should ideally include a comprehensive list of all allowed values for the input.
Please ensure that the implementation adheres to these guidelines.

## Issue

#68 

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/policy-hub/blob/main/docs/technical-documentation/dev-process/How%20to%20contribute.md)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added copyright and license headers, footers (for .md files) or files (for images)

### CODE REVIEW

File: src/database/PolicyHub.DbAccess/Repositories/PolicyRepository.cs [hunk 1]
```diff
@@ -79,4 +79,15 @@
...
             .AsAsyncEnumerable();
+    Task<bool> IPolicyRepository.CheckPolicyByTechnicalKeys(PolicyTypeId type, IEnumerable<string> technicalKeys) =>
+        _dbContext.Policies
+            .Where(p =>
+                p.Types.Any(t => t.IsActive && t.Id == type) &&
+                technicalKeys.Contains(p.TechnicalKey))
+            .AnyAsync();
+    public IAsyncEnumerable<string> GetAllTechnicalKeys() =>
+        _dbContext.Policies
+            .Where(x => x.IsActive == true)
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/database/PolicyHub.DbAccess/Repositories/PolicyRepository.cs:90]: @AnuragNagpure  please adjust according to the suggestion of CodeQL

File: src/database/PolicyHub.DbAccess/Repositories/PolicyRepository.cs [hunk 2]
```diff
@@ -79,4 +79,15 @@ public PolicyRepository(PolicyHubContext dbContext)
                     p.PolicyKind!.Configuration!.RightOperandValue
                 ))
             .AsAsyncEnumerable();
+    Task<bool> IPolicyRepository.CheckPolicyByTechnicalKeys(PolicyTypeId type, IEnumerable<string> technicalKeys) =>
+        _dbContext.Policies
+            .Where(p =>
+                p.Types.Any(t => t.IsActive && t.Id == type) &&
+                technicalKeys.Contains(p.TechnicalKey))
+            .AnyAsync();
```
Comment (Phil91) [inline src/database/PolicyHub.DbAccess/Repositories/PolicyRepository.cs:87]: Please add a line break in between those methods

**Review by Phil91 — CHANGES_REQUESTED** [review]

**Review by Phil91 — COMMENTED** [review]
With the current implementation only the left operand gets checked, the ticket mentions to check the value. at the moment there is no configuration for purposes to a given value set. I think we need to clarify with @jjeroch

**Comment by AnuragNagpure** [comment]
Below changes are added as per description.

PolicyAttributes

remove - ID Trace 3.1 from policy_attributes

map attribute_values:

purpose.trace.v1.traceBattery
purpose.trace.v1.aspects
purpose.trace.v1.qualityanalysis

to polcy_id = 01a0fba3-9b6e-435a-b045-e0e890c300c2

Policies

remove the following policies including the linked data

01a0fba3-9b6e-435a-b045-e0e890c300b7
01a0fba3-9b6e-435a-b045-e0e890c300b8
01a0fba3-9b6e-435a-b045-e0e890c300b9

add a check to the post endpoint to validate the value of the request to be existing in the attributes_value table for the given policy

/cc @jjeroch

**Review by Phil91 — APPROVED** [review]

**Pull request merged**